### PR TITLE
Implement "not to have (class|classes)" assertion.

### PR DESCRIPTION
**Pull request by alexjeffburke**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -724,6 +732,23 @@ module.exports = {
...
+
+        if (typeof value === 'string') {
+          value = getClassNamesFromAttributeValue(value);
+        }
+
+        value.forEach(namedValue =>
+          removeValueFromArray(expectedClasses, namedValue)
+        );
+
+        return expect(subject, 'to only have classes', expectedClasses);
```
Comment (papandreou) [inline src/index.js]: Hmm, couldn't you map it to:

```js
const classAttributeValue = getAttributes(subject).class;
const actualClasses = typeof classAttributeValue === 'string' ? getClassNamesFromAttributeValue(classAttributeValue) : [];
const expectedClasses = Array.isArray(value) ? value : [value];
expect(actualClasses, 'not to contain', ...expectedClasses);
```

That would produce a nicer error message and could also cover the `to have classes` case with a bit of flag forwarding?

Comment (alexjeffburke) [reply src/index.js]: Yeah could do - IIRC I did it this way for consistency something else in U-dom, I think it was https://github.com/unexpectedjs/unexpected-dom/blob/master/src/index.js#L1329. But happy to change it round if that works better :)

Comment (papandreou) [reply src/index.js]: When I looked at the test I thought it was confusing that it said:

> `class="bar quux" // expected [ 'bar', 'quux' ] to equal [ 'bar' ]`

... when the user didn't actually mention `bar` in the assertion.

If the other place is equivalent to that, I think we should change that as well :hugs:

Comment (alexjeffburke) [reply src/index.js]: Have pushed a commit to make the assertion run via "not to contain" :)

Comment (papandreou) [reply src/index.js]: Yay, thanks! :tada:

File: src/index.js [hunk 2]
```diff
@@ -99,6 +99,14 @@ function validateStyles(expect, str) {
   }
 }
 
+function removeValueFromArray(array, value) {
```
Comment (papandreou) [inline src/index.js]: Can be removed now

Comment (alexjeffburke) [reply src/index.js]: Argh! Knew I left something - thanks!

**Comment by sunesimonsen** [comment]
Don't er need docs?

**Comment by alexjeffburke** [comment]
@sunesimonsen you were absolutely right and that should have been in there :)

**Comment by alexjeffburke** [comment]
This closes #201

**Pull request merged**